content/docs/source-git/work-with-source-git/_index.md

```diff
@@ -0,0 +1,112 @@
+
+The key ingredient here is the config option `upstream_ref`. It tells packit
+the git ref which is supposed to match the archive specified in the spec file.
+All commits on top of this ref are treated either as:
+
+* downstream patches
+* downstream packaging
+
+For downstream patches, packit creates a patch file for each commit on top of
+`upstream_ref`. Downstream packaging changes are ignored in patch files since
+they can be directly committed to dist-git. You can control the ignore
+mechanism with config option `patch_generation_ignore_paths`.
+
+
+### Example
+
+Let's have a look at source-git repo for pacemaker package:
+
+    $ git log --online
+    bd12722 (HEAD -> c8s-test, origin/c8s-test, master) downstream packaging
+    7201e28 Refactor: controller: remove unused argument
+    91c557c Refactor: controller: convert active_op_t booleans to bitmask
+    2e90063 Refactor: controller: rename struct recurring_op_s to active_op_t
+    3ee90ce (tag: source-git-start) pacemaker-2.0.3 base
+
+We have 5 commits:
+* bd12722 - top commit, contains a spec file so packit won't create a patch file out of this one
+* 7201e28 - the 3rd downstream patch
+* 91c557c - the 2nd downstream patch
+* 2e90063 - the 1st downstream patch
+* 3ee90ce - the bottom commit, equals to an unpacked upstream tarball
+
+packit.yaml has only two lines:
+
+    $ cat .packit.yaml
+    specfile_path: SPECS/pacemaker.spec
+    upstream_ref: source-git-start
+
+As you can see, we are telling packit, that commit tagged as `source-git-start`
+indicates the barrier between upstream and downstream, in our case, it's just a
+single commit with the upstream archive content, but it could have been the
+whole upstream git history.
+
+
+### Rebase or merge?
+
+A common workflow in the world of open source development is to accept
+pull requests and have merge commits in the git history. If the PRs are not
+rebased against the main branch before the merge, the history may go wild.
+Packit is able to work with such a state by creating an ephemeral branch with
+linear history and generate the patch files out of it. [More
+info](https://github.com/packit/packit/pull/766).
+
+Picking up latest upstream changes into your downstream source-git repo has
+multiple solutions and it's up to you to pick the one which suits your workflow
+best:
+
+1. **Rebase against the main upstream branch** (a.k.a. `git pull --rebase
+   upstream master`). This solution implies that you are able to perform the
+   force-push operation (e.g. Fedora and CentOS dist-git instances don't allow
+   force-pushes). Packit will not help you with the rebase process in any way
+   and it's up to you to complete the rebase.
+2. **Create a new branch for every upstream release**. No rebase needed — may
+   require good branch hygiene since the number of branches can skyrocket. No
+   dedicated help from packit again, you are running all the `git` commands.
```

Comment:
Wondering if there would be any "hand-holding" that Packit could do here? :thinking:

Reply:
We could certainly create an automation for this, that would be pretty awesome.